models/utilities/default_model_mod.f90

```diff
@@ -283,6 +300,162 @@ subroutine pert_model_copies(state_ens_handle, ens_size, pert_amp, interf_provid
 
 end subroutine pert_model_copies
 
+!--------------------------------------------------------------------
+
+!> Reads in model_nml:model_state_variables and returns a
+!> state_var_type state_vars with nvars ; netcdf variable names ;
+!> qtys (kinds) ; clamp values ; updates
+!
+!> Verifies that the namelist was filled in correctly, and checks
+!> that there are valid entries for the dart_kind.
+
+subroutine get_state_variables_clamp(nml_state_vars, MAX_STATE_VARIABLES, use_clamping, state_vars)
```

Comment:
Can this be a function not a subroutine?

state_vars = get_state_variables_clamp(..)